qubes/app.py

```diff
@@ -1404,6 +1404,22 @@ def _get_pool(**kwargs):
             raise qubes.exc.QubesException('No driver %s for pool %s' %
                                            (driver, name))
 
+    async def stop_storage(self):
+        """
+        Stop the storage of all domains that are not running.
+        """
+        async def stop(i):
+            async with i.startup_lock:
+                if not i.is_running():
+                    await i.storage.stop()
+        future = tuple(stop(i) for i in self.domains if i.klass != 'AdminVM')
+        finished = ()
+        while future:
+            qubes.utils.systemd_extend_timeout()
+            finished, future = await asyncio.wait(future, timeout=30000000)
+            for i in finished:
+                await i
```

Comment:
This will interrupt processing if _any_ stop fails. Two important questions here:
1. Should it keep waiting for other calls to finish instead (currently it wont, if they take more time than the current iteration)? I think yes.
2. Should the exception prevent qubesd startup (it will now)? I think it shouldn't. Generally failing qubesd startup will make the system mostly unusable. If you use USB keyboard+mouse and sys-usb, then literally unusable. On the other hand, the very VM that has failed storage stop, most likely be in a weird state. In any case, having one broken VM should not prevent using others.

Reply:
Perhaps `.start()` should include an implicit `.stop()`?  I really don’t want to be able to start a qube if it is in a broken state.

Reply:
>  I really don’t want to be able to start a qube if it is in a broken state.

That isn't really a concern. `start()` should be able to deal with a semi-broken state (like - interrupted stop), or raise an exception if it cannot. After all, before this PR, `start()` was a normal path to recovery from a hard system reset.